### PR TITLE
Core/PacketIO: updated some pet packets

### PR DESCRIPTION
**Pull request by ForesterDev**
**Changes proposed:**

-  Update pet packets: CMSG_PET_ABANDON , CMSG_PET_STOP_ATTACK , CMSG_PET_SPELL_AUTOCAST, SMSG_PET_LEARNED_SPELL , SMSG_PET_UNLEARNED_SPELL

*Target branch(es):** 3.3.5/master

- [x] 3.3.5

**Tests performed:** (Does it build, tested in-game, etc.)
Build, doesn't tested in game (maybe someone can test)

### CODE REVIEW

File: src/server/game/Handlers/PetHandler.cpp [hunk 1]
```diff
@@ -732,22 +717,22 @@ void WorldSession::HandlePetSpellAutocastOpcode(WorldPacket& recvPacket)
...
         if (!charmInfo)
         {
-            TC_LOG_ERROR("entities.pet", "WorldSession::HandlePetSpellAutocastOpcod: object %s is considered pet-like but doesn't have a charminfo!", petControlled->GetGUID().ToString().c_str());
+            TC_LOG_ERROR("entities.pet", "WorldSession::HandlePetSpellAutocastOpcode: object %s is considered pet-like but doesn't have a charminfo!", petControlled->GetGUID().ToString().c_str());
             return;
         }
 
         if (petControlled->IsPet())
-            ((Pet*)petControlled)->ToggleAutocast(spellInfo, state != 0);
+            petControlled->ToPet()->ToggleAutocast(spellInfo, packet.AutocastEnabled);
```
Comment (soulfrost) [inline src/server/game/Handlers/PetHandler.cpp:731]: if (Pet* pet = petControlled->ToPet())
      pet ->ToggleAutocast(spellInfo, packet.AutocastEnabled);

File: src/server/game/Server/Packets/PetPackets.h [hunk 2]
```diff
@@ -35,6 +35,58 @@ namespace WorldPackets
...
+        };
+
+        class PetUnlearnedSpell final : public ServerPacket
+        {
+        public:
+            PetUnlearnedSpell() : ServerPacket(SMSG_PET_UNLEARNED_SPELL, 4) { }
+
+            WorldPacket const* Write() override;
+
+            uint32 SpellID;
```
Comment (soulfrost) [inline src/server/game/Server/Packets/PetPackets.h:87]: uint32 SpellID = 0;

File: src/server/game/Server/Packets/PetPackets.h [hunk 3]
```diff
@@ -35,6 +35,58 @@ namespace WorldPackets
...
+        };
+
+        class PetLearnedSpell final : public ServerPacket
+        {
+        public:
+            PetLearnedSpell() : ServerPacket(SMSG_PET_LEARNED_SPELL, 4) { }
+
+            WorldPacket const* Write() override;
+
+            uint32 SpellID;
```
Comment (soulfrost) [inline src/server/game/Server/Packets/PetPackets.h:77]: uint32 SpellID = 0;

File: src/server/game/Server/Packets/PetPackets.h [hunk 4]
```diff
@@ -35,6 +35,58 @@ namespace WorldPackets
...
+
+        class PetSpellAutocast final : public ClientPacket
+        {
+        public:
+            PetSpellAutocast(WorldPacket&& packet) : ClientPacket(CMSG_PET_SPELL_AUTOCAST, std::move(packet)) { }
+
+            void Read() override;
+
+            ObjectGuid PetGUID;
+            uint32 SpellID;
```
Comment (soulfrost) [inline src/server/game/Server/Packets/PetPackets.h:66]: uint32 SpellID = 0;
bool AutocastEnabled = false;

**Comment by jackpoz** [comment]
Thanks for the PR :)

**Pull request merged**